numpy/core/tests/test_overrides.py

```diff
@@ -514,15 +504,27 @@ def call(func, args, kwargs, like=None):
          ('1,2',),
          {'dtype': int, 'sep': ','}),
         ('loadtxt',
-         "(StringIO('0 1\\n2 3'),)",
+         None,
          {}),
         ('genfromtxt',
-         "(StringIO(u'1,2.1'),)",
+         None,
          {'dtype': [('int', 'i8'), ('float', 'f8')], 'delimiter': ','}),
         ])
     @pytest.mark.parametrize('numpy_ref', [True, False])
     @requires_array_function
-    def test_array_like_clean(self, function, args, kwargs, numpy_ref):
+    def test_array_like(self, function, args, kwargs, numpy_ref):
+        def _get_data():
+            # Pytest will not parametrize objects multiple times, due
+            # to that StringIO objects will be "consumed" by the first
+            # test and fail after, thus we create new objects for
+            # each usage.
+            if function == 'loadtxt':
+                return (StringIO('0 1\n2 3'),)
+            elif function == 'genfromtxt':
+                return (StringIO(u'1,2.1'),)
+            else:
+                return args
```

Comment:
Can you parametrize over a lambda instead of this?

Reply:
That's a great idea, thanks @hameerabbasi ! I've pushed changes for that now.